authentication/views.py

```diff
@@ -0,0 +1,33 @@
+""" All Views of the Authentication Application
+
+This file contains all the views of the authentication Application. The views
+are used to create, update, delete and list the users.
+
+"""
+
+# pylint: disable=no-member,too-many-ancestors
+from rest_framework import generics, status
+from rest_framework.response import Response
+from authentication.serializers import UserRegisterSerializer
+
+
+class UserRegisterView(generics.GenericAPIView):
+    """User Registration View
+
+    This view is used to create a new user. The user is created using the
+    GenericAPIView class. The serializer is used to validate the data. The user
+    is created and the data is returned.
+    """
+
+    serializer_class = UserRegisterSerializer
+
+    def post(self, request):
```

Comment:
I don't think we should do all this (override post and add our own code etc) since we can use `CreateModelMixin` as mentioned above